topics/lua-api.md

```diff
@@ -383,86 +383,116 @@ This function prints its argument in the [Redis Lua debugger](/topics/ldb) conso
 
 This function is only available from the context of the `FUNCTION LOAD` command.
 When called, it registers a function to the loaded library.
-The function can be called either with positional argument or with named arguments.
+The function can be called either with positional or named arguments.
 
 #### <a name="redis.register_function_pos_args"></a> positional arguments: `redis.register_functio(name, callback)`
 
-The first argument to `redis.register_functio` is a Lua string representing the function name.
-The second argument to `redis.register_functio` is a Lua function.
+The first argument to `redis.register_function` is a Lua string representing the function name.
+The second argument to `redis.register_function` is a Lua function.
 
-Usage Example:
+Usage example:
 
 ```
 redis> FUNCTION LOAD Lua mylib "redis.register_function('noop', function() end)"
 ```
 
 #### <a name="redis.register_function_named_args"></a> Named arguments:  `redis.register_functio{function_name=name, callback=callback, flags={flag1, flag2, ..}. description=description}`
 
-The named argument version except the following named arguments:
+The named arguments variant accepts the following arguments:
 
-* _function\_name_ - Lua string representing the function name
-* _callback_ - Lua function
-* _flags_ - Lua array of strings representing function flags
-* _description_ - Lua string representing the function description
+* _function\_name_: the function's name.
+* _callback_: the function's callback.
+* _flags_: an array of strings, each a function flag (optional).
+* _description_: function's description (optional).
 
-_function\_name_ and _callback_ are mandatory, the other named arguments are option.
+Both _function\_name_ and _callback_ are mandatory.
 
-Usage Example:
+Usage example:
 
 ```
-redis> FUNCTION LOAD Lua mylib "redis.register_function{function_name='noop', callback=function() end, function, flags={'no-writes'}, description='some desc')"
+redis> FUNCTION LOAD Lua mylib "redis.register_function(function_name='noop', callback=function() end, flags={ 'no-writes' }, description='Does nothing')"
 ```
 
-#### <a name="script_flags"></a> Script Flags
+#### <a name="function_flags"></a> Function flags
```

Comment:
flags are now serving EVAL scripts too, so maybe we should call the section "script flags"
and have a short intro that states (links) how you apply them to either Eval or Function.

we can maybe leave the details about EVAL to be added later.
but let's avoid calling them "function flags" in all the text below, and the section name.